app/lib/actions.ts

```diff
@@ -0,0 +1,35 @@
+import { Response } from "@/app/lib/definitions";
+
+const PUBLIC_BASE_URL =
+  process.env.PUBLIC_BASE_URL ||
+  "https://f442285821a31af458af8b09d237e087.serveo.net";
+
+const CREDENTIAL_DEFINITION_ID =
+  process.env.CREDENTIAL_DEFINITION_ID ||
+  "did:web:chatbot-demo.dev.2060.io?service=anoncreds&relativeRef=/credDef/HngJhYMeTLTZNa5nJxDybmXDsV8J7G1fz2JFSs3jcouT";
+
+export async function getQR(socketConnectionId: string): Promise<Response> {
```

Comment:
This code must run on the backend side (i.e. create an api/getqr endpoint that takes the socketConnectionId as body parameter, and performs this fetch to the Service Agent instance).